plutus-core/plutus-core/src/PlutusCore/Constant/Debug.hs

```diff
@@ -46,8 +46,9 @@ enumerateDebug = id
 -- inferDebug False :: Bool
 -- >>> :t inferDebug $ Just 'a'
 -- <interactive>:1:1-21: error:
---     • There's no 'KnownType' instance for Maybe Char
---       Did you add a new built-in type and forget to provide a 'KnownType' instance for it?
+--     • No instance for (KnownPolytype
```

Comment:
Oh shit, I did not realize that while we don't need to give this error message about a new built-in type anymore (because it's gonna be triggered way earlier), we probably still want to say to the user that a certain type (like `Maybe Char`) is not supported in the type of a built-in function. I'll try to reintroduce the `OVERLAPPABLE` instance, now that we have explicit type application in that weird Plutus program generation code, maybe it's gonna work just fine.

Reply:
Nope, reintroducing the `OVERLAPPABLE` instance results in conflicting `AssociateType` definitions, so the instance does not type check. I'll think about this problem more, but it shouldn't block this PR.